.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,33 +1,42 @@
-## Description
+<!-- Note: replace the instructions with your text -->
 
-<!-- Just describe what you did. -->
+## Context
 
-<!-- Skip if the title of the PR is self-explanatory -->
+- Describe the objective or issue this PR addresses, as well as the impact of the change.
+- Try to keep the description accessible to newcomers.
+- If you're resolving a specific issue, add "Fixes #issue_number" or "Closes #issue_number".
 
-### Linked issue
+### Solution
 
-<!-- Duplicate the main issue and add additional issues closed by this PR. -->
+- Describe the approach taken to achieve the objective / resolve the issue.
 
-Closes #{issue_number} <!-- Replace with an actual number,  -->
+### Migration Guide (optional)
 
-<!-- Link if e.g. JIRA issue or  from another repository -->
+- If this PR contains a breaking change relative to the `main` branch, provide an instruction on how affected parties might need to adapt to the change.
 
-### Benefits
+---
 
-<!-- EXAMPLE: users can't revoke their own right to revoke rights -->
+### Review notes (optional)
+
+- For complex PRs, try to provide some information on how to approach the review more effectively.
+- For example, is there a natural order in which the affected files should be reviewed?
+
+> #### On review etiquette
+> 
+> - Do not resolve a conversation on your own. Let the reviewer make a decision.
+> - Acknowledge review comments and engage with the reviewer (agree, disagree, clarify, explain, etc.). Do not ignore comments.
+> - For simple code change suggestions, if you apply them directly, you can resolve the conversation.
```

Comment:
Should this be part of `CONTRIBUTING.md`?

Reply:
Moved it